File: readability/readability.py

from cs50 import get_string

# get user input
text = get_string("Text: ")

# initialize variables
letters = 0
words = 1
sentences = 0
length = len(text)

for i in range(length):

    # count the letters
    if (text[i].isalpha()):
        letters += 1

    # count the words
    if (text[i].isspace()):
        words += 1

    # count the sentences
    if (text[i] == '.' or text[i] == '!' or text[i] == '?'):
        sentences += 1

# calculate the Coleman-Liau Index
L = (letters / float(words)) * 100
S = (sentences / float(words)) * 100
grade = round(0.0588 * L - 0.296 * S - 15.8)
# print out the appropriate grade
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade: {grade}")

File: readability/readability.c
# include <cs50.h>
# include <stdio.h>
# include <ctype.h>
# include <math.h>
# include <string.h>

int main(void)
{
    // get text from user
    string text = get_string("Text: ");

    //initialise letters, words and sentences

    int letters = 0;
    int words = 1;
    int sentences = 0;

    //count letters

    for (int i = 0, n = strlen(text); i < n; i++)
     while (text[i] != '\0')
      {
         if (isalpha(text[i++]))
          {
              letters++;
          }
      }
    {
        printf("Letters: %i\n", letters);
    }

    //count words

    for(int i = 0, n = strlen(text); i < n; i++)
    {

        if ((text[i] == 0 && text[i] != ' ') || (text[i] != n-1 && text[i] == ' ' && text[i+1] != ' ' ))
        {
          words++;
         }
    }
    {
        printf("Words: %i\n", words);
    }

    //count sentences

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    {
        printf("Sentences: %i\n", sentences);
    }

    //calculate average letters and sentences and cast int to float
    float L = (letters / (float) words) * 100;
    float S = (sentences / (float) words) * 100;

// calculate coleman liau index

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 1 && index < 16)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }

}
